### DISCUSSION TITLE
writeRaster for LAScatalog

### DISCUSSION THREAD

**Discussion by taeyoonlee87** [r-lidar/lidR#654] [Q&A] [answered]
I got an error message “Error in (function (classes, fdef, mtable) : unable to find an inherited method for function ‘writeRaster’ for signature ‘"LAScatalog", "character" “.. 

It seems like I cannot use writeRaster for LAScatalog. but this error makes me confused. 

Here is my code. 

```r
setwd("E:/Lidar_2022/R/Data_R")
las_cat<-readLAScatalog("E:/Forest_LiDAR/226776/tiles")
opt_output_files(las_cat) <- "E:/Lidar_2022/R/Data_R2/dtm_{XLEFT}_{YBOTTOM}"
dtm <- grid_terrain(las_cat, res = 2, algorithm = knnidw(k=10, p=2, rmax=50), keep_lowest = FALSE)
writeRaster(dtm, "E:/Lidar_2022/R/Data_R/dtm.tif", overwrite=TRUE)                                                              **<- no problem in here**

opt_output_files(las_cat) <-"E:/Lidar_2022/R/Data_R2/norm_{XLEFT}_{YBOTTOM}"
MyNorm <- normalize_height(las_cat, dtm) 
writeRaster(MyNorm, "E:/Lidar_2022/R/Data_R/chm.tif", overwrite=TRUE)                                                          **<- error** 
#> Error in (function (classes, fdef, mtable) : unable to find an inherited method for function ‘writeRaster’ for signature ‘"LAScatalog", "character" 
```

The outcome from `grid_terrain` was able to be written as raster (`.tif` file) but the outcome from `normalize_height` was not able to be used by `writeRaster` 

what is the difference between the two outcomes? I believe both are LAScatalog. 

And how can I bring back the "dtm" (LAS catalog) from raster (dtm.tif) without re-running `grid_terrain`?

When I used a function of  `raster`, it seems like `RasterLayer` instead of las file or `LAScatalog`

Thanks in advance!

**Comment by Jean-Romain** [comment, marked as answer]
- `grid_terrain` (deprecated, use `rasterize_terrain`) produces a DTM i.e. a raster. A raster can indeed be written in a raster format
- `normalize_height` produces a point cloud. A point cloud is not a raster and cannot be written in a raster format
- considering you used `opt_output_file` to produced the DTM you do not need to use `writeRaster`. The raster is already saved on disk.
- A `LAScatalog` is an object handling a collection of las file on disk.
- `rasterize_canopy` produces a CHM.

**Reply by taeyoonlee87** [reply]
Thank you for replying to me. 
It makes me clear! 

By the way, I am still wondering how I can get LAS file from raster (*.tif) without running "grid_terrain"?

**Reply by Jean-Romain** [reply]
I don't understand your question. A tif file is a raster. A las file is a point cloud. You can't have a point cloud from a raster. 
Well actually you can consider each pixel like point and have a regularly spaced point cloud but lidR does not offer such tool out of the box.
So you can't *"get a LAS file from a raster"*

**Reply by taeyoonlee87** [reply]
Ah,,, so the outcome(dtm) from “grid-terrain” is a LAScatalog. 
Then, is there anyway to save LAS catalog as a single las file so that I can save it and retrieve whenever I need.
I hope my question is makes sense as I am a newbie in analyzing LiDAR.

**Reply by Jean-Romain** [reply]
> Ah, so the outcome(dtm) from “grid-terrain” is a LAScatalog.

No it is a raster. `grid_terrain` rasterizes the point cloud to produce a raster (here a dtm)

> Then, is there anyway to save LAS catalog as a single las file so that I can save it and retrieve whenever I need.

A LAScatalog is an object pointing to multiple LAS files. You could merge all your LAS files into one huge file but what would be the purpose?

**Reply by taeyoonlee87** [reply]
I see now, 

I really appreciate your response.
